nbclassic/notebookapp.py

```diff
@@ -218,6 +219,17 @@ def initialize_handlers(self):
         """Load the (URL pattern, handler) tuples for each component."""
         # Order matters. The first handler to match the URL will handle the request.
         handlers = []
+
+        # Add a redirect from /notebooks to /edit
+        # for opening non-ipynb files in edit mode.
+        handlers.append(
+            (
+                rf"/({self.file_url_prefix})/(.*)\.(((?!ipynb)).*)",
```

Comment:
This regex might need some tweaking to also handle files without a file extension, for example:

```bash
jupyter nbclassic LICENSE
```

Reply:
Are you waiting on this or is the PR ready for merge?

Reply:
It would be fine to merge as is, and we can open a new issue to track this.